### ISSUE TITLE
Have a look at the static warnings and errors currently reported

### ISSUE THREAD

**Issue by jurgenvinju** [cwi-swat/php-analysis#15]
This is not immediately obvious:
* They could be actual issues
* They could be false reports by the checker
* They could be caused by wrongly configured search paths for the checker

```
[INFO] checker is done, reporting errors now.
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/analysis/cfg/BuildCFG.rsc|
Warning:  lang/php/analysis/cfg/BuildCFG.rsc:0105:000: Unused function `getScriptFunctions`
Warning:  lang/php/analysis/cfg/BuildCFG.rsc:0098:000: Unused function `getScriptMethods`
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/analysis/signatures/Summaries.rsc|
Error:    lang/php/analysis/signatures/Summaries.rsc:0017:007: Reference to name `lang::php::util::Config` cannot be resolved
Warning:  lang/php/analysis/signatures/Summaries.rsc:0017:000: Unused import of `lang::php::util::Config`
Error:    lang/php/analysis/signatures/Summaries.rsc:0065:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/analysis/signatures/Summaries.rsc:0017:007: Undefined module `lang::php::util::Config`
Error:    lang/php/analysis/signatures/Summaries.rsc:0073:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/analysis/signatures/Summaries.rsc:0049:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/analysis/signatures/Summaries.rsc:0057:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/analysis/signatures/Summaries.rsc:0016:007: Reference to name `ValueIO` cannot be resolved
Error:    lang/php/analysis/signatures/Summaries.rsc:0069:008: Undefined  `readBinaryValueFile`
Error:    lang/php/analysis/signatures/Summaries.rsc:0053:008: Undefined  `readBinaryValueFile`
Warning:  lang/php/analysis/signatures/Summaries.rsc:0016:000: Unused import of `ValueIO`
Error:    lang/php/analysis/signatures/Summaries.rsc:0045:008: Undefined  `readBinaryValueFile`
Error:    lang/php/analysis/signatures/Summaries.rsc:0016:007: Undefined module `ValueIO`
Error:    lang/php/analysis/signatures/Summaries.rsc:0061:008: Undefined  `readBinaryValueFile`
Error:    lang/php/analysis/signatures/Summaries.rsc:0019:027: Undefined  `baseLoc`
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/analysis/callgraph/SimpleCG.rsc|
[INFO]    lang/php/analysis/callgraph/SimpleCG.rsc:0119:053: Pattern variable `cn` has been declared outside pattern and its value will be used, add explicit declaration here if you want a new variable
[INFO]    lang/php/analysis/callgraph/SimpleCG.rsc:0119:056: Pattern variable `mn` has been declared outside pattern and its value will be used, add explicit declaration here if you want a new variable
Warning:  lang/php/analysis/callgraph/SimpleCG.rsc:0016:000: Unused import of `lang::php::analysis::NamePaths`
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/analysis/includes/IncludesInfo.rsc|
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0138:041: Undefined  `readBinaryValueFile`
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0028:007: Undefined module `ValueIO`
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0140:027: Undefined  `readBinaryValueFile`
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0082:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0139:036: Undefined  `readBinaryValueFile`
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0141:042: Undefined  `readBinaryValueFile`
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0017:007: Undefined module `lang::php::util::Config`
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0028:007: Reference to name `ValueIO` cannot be resolved
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0085:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0084:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0031:022: Undefined  `baseLoc`
Warning:  lang/php/analysis/includes/IncludesInfo.rsc:0028:000: Unused import of `ValueIO`
Warning:  lang/php/analysis/includes/IncludesInfo.rsc:0017:000: Unused import of `lang::php::util::Config`
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0083:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/analysis/includes/IncludesInfo.rsc:0017:007: Reference to name `lang::php::util::Config` cannot be resolved
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/util/CLOC.rsc|
Warning:  lang/php/util/CLOC.rsc:0028:023: Unused pattern variable `n2`
Warning:  lang/php/util/CLOC.rsc:0028:012: Unused pattern variable `n1`
Warning:  lang/php/util/CLOC.rsc:0028:034: Unused pattern variable `n3`
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/stats/SLOC.rsc|
Error:    lang/php/stats/SLOC.rsc:0029:014: Undefined  `countsDir`
Error:    lang/php/stats/SLOC.rsc:0015:007: Reference to name `lang::php::util::Config` cannot be resolved
Error:    lang/php/stats/SLOC.rsc:0015:007: Undefined module `lang::php::util::Config`
Warning:  lang/php/stats/SLOC.rsc:0015:000: Unused import of `lang::php::util::Config`
Error:    lang/php/stats/SLOC.rsc:0026:013: Undefined  `countsDir`
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/analysis/cfg/Label.rsc|
Warning:  lang/php/analysis/cfg/Label.rsc:0016:000: Unused import of `lang::php::ast::AbstractSyntax`
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/util/Corpus.rsc|
Error:    lang/php/util/Corpus.rsc:0054:038: Undefined  `corpusRoot`
Error:    lang/php/util/Corpus.rsc:0034:025: Undefined  `corpusRoot`
Error:    lang/php/util/Corpus.rsc:0017:007: Undefined module `lang::php::util::Config`
Error:    lang/php/util/Corpus.rsc:0190:037: Undefined  `corpusRoot`
Error:    lang/php/util/Corpus.rsc:0017:007: Reference to name `lang::php::util::Config` cannot be resolved
Error:    lang/php/util/Corpus.rsc:0040:025: Undefined  `corpusRoot`
Error:    lang/php/util/Corpus.rsc:0037:005: Undefined  `useBinaries`
Error:    lang/php/util/Corpus.rsc:0052:021: Undefined  `parsedDir`
Error:    lang/php/util/Corpus.rsc:0045:029: Undefined  `parsedDir`
Error:    lang/php/util/Corpus.rsc:0073:021: Undefined  `corpusRoot`
Error:    lang/php/util/Corpus.rsc:0091:021: Undefined  `corpusRoot`
Error:    lang/php/util/Corpus.rsc:0077:017: Undefined  `corpusRoot`
Error:    lang/php/util/Corpus.rsc:0079:017: Undefined  `corpusRoot`
Error:    lang/php/util/Corpus.rsc:0033:030: Undefined  `baseLoc`
Error:    lang/php/util/Corpus.rsc:0093:017: Undefined  `corpusRoot`
Error:    lang/php/util/Corpus.rsc:0095:017: Undefined  `corpusRoot`
Error:    lang/php/util/Corpus.rsc:0044:005: Undefined  `useBinaries`
Error:    lang/php/util/Corpus.rsc:0051:005: Undefined  `useBinaries`
Error:    lang/php/util/Corpus.rsc:0097:017: Undefined  `corpusRoot`
Error:    lang/php/util/Corpus.rsc:0038:021: Undefined  `parsedDir`
Error:    lang/php/util/Corpus.rsc:0047:044: Undefined  `corpusRoot`
Warning:  lang/php/util/Corpus.rsc:0017:000: Unused import of `lang::php::util::Config`
Error:    lang/php/util/Corpus.rsc:0075:017: Undefined  `corpusRoot`
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/analysis/signatures/Extract.rsc|
Warning:  lang/php/analysis/signatures/Extract.rsc:0107:007: Unused pattern variable `class`
Warning:  lang/php/analysis/signatures/Extract.rsc:0078:015: Unused pattern variable `funnamelink`
Warning:  lang/php/analysis/signatures/Extract.rsc:0103:012: Unused pattern variable `classlinkname`
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/util/RepoUtils.rsc|
Error:    lang/php/util/RepoUtils.rsc:0015:007: Reference to name `util::git::Git` cannot be resolved
Warning:  lang/php/util/RepoUtils.rsc:0015:000: Unused import of `util::git::Git`
Error:    lang/php/util/RepoUtils.rsc:0022:008: Undefined  `switchToTag`
Error:    lang/php/util/RepoUtils.rsc:0030:008: Undefined  `switchToTag`
Error:    lang/php/util/RepoUtils.rsc:0020:011: Undefined  `getTags`
Error:    lang/php/util/RepoUtils.rsc:0028:011: Undefined  `getTags`
Error:    lang/php/util/RepoUtils.rsc:0037:011: Undefined  `getTags`
Error:    lang/php/util/RepoUtils.rsc:0039:008: Undefined  `switchToTag`
Error:    lang/php/util/RepoUtils.rsc:0015:007: Undefined module `util::git::Git`
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/stats/Stats.rsc|
Error:    lang/php/stats/Stats.rsc:0026:007: Reference to name `ValueIO` cannot be resolved
Error:    lang/php/stats/Stats.rsc:0462:013: Undefined  `statsDir`
Error:    lang/php/stats/Stats.rsc:0027:007: Reference to name `lang::csv::IO` cannot be resolved
Error:    lang/php/stats/Stats.rsc:0473:013: Undefined  `statsDir`
Error:    lang/php/stats/Stats.rsc:0556:054: Undefined  `readCSV`
Warning:  lang/php/stats/Stats.rsc:0026:000: Unused import of `ValueIO`
Error:    lang/php/stats/Stats.rsc:0026:007: Undefined module `ValueIO`
Error:    lang/php/stats/Stats.rsc:0463:013: Undefined  `statsDir`
Warning:  lang/php/stats/Stats.rsc:0020:000: Unused import of `lang::php::util::Config`
Error:    lang/php/stats/Stats.rsc:0498:088: Undefined  `readCSV`
Error:    lang/php/stats/Stats.rsc:0020:007: Undefined module `lang::php::util::Config`
Error:    lang/php/stats/Stats.rsc:0458:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Stats.rsc:0467:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Stats.rsc:0465:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Stats.rsc:0472:013: Undefined  `statsDir`
Error:    lang/php/stats/Stats.rsc:0474:052: Undefined  `readBinaryValueFile`
Warning:  lang/php/stats/Stats.rsc:0027:000: Unused import of `lang::csv::IO`
Error:    lang/php/stats/Stats.rsc:0504:060: Undefined  `readCSV`
Error:    lang/php/stats/Stats.rsc:0027:007: Undefined module `lang::csv::IO`
Error:    lang/php/stats/Stats.rsc:0457:013: Undefined  `statsDir`
Error:    lang/php/stats/Stats.rsc:0466:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Stats.rsc:0471:013: Undefined  `statsDir`
Error:    lang/php/stats/Stats.rsc:0474:010: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Stats.rsc:0474:094: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Stats.rsc:0020:007: Reference to name `lang::php::util::Config` cannot be resolved
Error:    lang/php/stats/Stats.rsc:0464:013: Undefined  `statsDir`
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/stats/Unfriendly.rsc|
Error:    lang/php/stats/Unfriendly.rsc:0994:008: Undefined  `readBinaryValueFile`
Warning:  lang/php/stats/Unfriendly.rsc:0020:000: Unused import of `ValueIO`
Error:    lang/php/stats/Unfriendly.rsc:0700:008: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0560:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0023:007: Undefined module `Type`
Error:    lang/php/stats/Unfriendly.rsc:0563:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1748:032: Undefined constructor `\tuple`
Error:    lang/php/stats/Unfriendly.rsc:1114:026: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:0561:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0564:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1496:025: Undefined constructor `\set`
Error:    lang/php/stats/Unfriendly.rsc:0565:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1744:008: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0566:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1367:052: No definition for `symbol` in type `type[rel[str, str, str, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int]]`
Error:    lang/php/stats/Unfriendly.rsc:1616:013: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:1423:017: Undefined constructor `\map`
[INFO]    lang/php/stats/Unfriendly.rsc:1716:029: Pattern variable `version` has been declared outside pattern and its value will be used, add explicit declaration here if you want a new variable
Error:    lang/php/stats/Unfriendly.rsc:1[822](https://github.com/cwi-swat/php-analysis/actions/runs/6271280617/job/17030583653#step:5:823):001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0181:046: Undefined  `baseLoc`
Warning:  lang/php/stats/Unfriendly.rsc:0509:028: Unused pattern variable `p`
Error:    lang/php/stats/Unfriendly.rsc:1423:049: Undefined constructor `Type::label`
Error:    lang/php/stats/Unfriendly.rsc:0581:001: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1604:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1423:040: No definition for `symbol` in type `type[map[str, tuple[int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int]]]`
Error:    lang/php/stats/Unfriendly.rsc:0588:001: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0166:008: Undefined  `readCSV`
Error:    lang/php/stats/Unfriendly.rsc:2122:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1931:001: Undefined  `writeBinaryValueFile`
Warning:  lang/php/stats/Unfriendly.rsc:0023:000: Unused import of `Type`
Warning:  lang/php/stats/Unfriendly.rsc:0037:000: Unused import of `lang::csv::IO`
Error:    lang/php/stats/Unfriendly.rsc:0165:040: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:0023:007: Reference to name `Type` cannot be resolved
Error:    lang/php/stats/Unfriendly.rsc:0559:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0585:001: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0586:001: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0020:007: Undefined module `ValueIO`
Error:    lang/php/stats/Unfriendly.rsc:1742:014: Undefined  `parsedDir`
Error:    lang/php/stats/Unfriendly.rsc:1621:008: Undefined  `readBinaryValueFile`
Warning:  lang/php/stats/Unfriendly.rsc:2059:044: Unused pattern variable `cn`
Error:    lang/php/stats/Unfriendly.rsc:1367:026: Undefined constructor `\tuple`
Error:    lang/php/stats/Unfriendly.rsc:1617:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0032:007: Undefined module `lang::php::util::Config`
Error:    lang/php/stats/Unfriendly.rsc:1117:059: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:0696:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1367:021: Undefined constructor `\set`
Error:    lang/php/stats/Unfriendly.rsc:1748:058: No definition for `symbol` in type `type[rel[str, str, str, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int]]`
Error:    lang/php/stats/Unfriendly.rsc:1011:018: Undefined constructor `Type::label`
Error:    lang/php/stats/Unfriendly.rsc:1521:031: Undefined constructor `\set`
Error:    lang/php/stats/Unfriendly.rsc:0032:007: Reference to name `lang::php::util::Config` cannot be resolved
Error:    lang/php/stats/Unfriendly.rsc:0174:008: Undefined  `readCSV`
Error:    lang/php/stats/Unfriendly.rsc:0580:001: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:2114:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1603:059: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:0589:001: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0099:048: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:1600:029: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:2319:008: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1768:001: Undefined  `writeBinaryValueFile`
Warning:  lang/php/stats/Unfriendly.rsc:0032:000: Unused import of `lang::php::util::Config`
Error:    lang/php/stats/Unfriendly.rsc:0777:008: Undefined  `readBinaryValueFile`
[INFO]    lang/php/stats/Unfriendly.rsc:2218:042: Pattern variable `v` has been declared outside pattern and its value will be used, add explicit declaration here if you want a new variable
Warning:  lang/php/stats/Unfriendly.rsc:1085:052: Unused formal `label`
Error:    lang/php/stats/Unfriendly.rsc:0587:001: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0772:013: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:0562:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0037:007: Undefined module `lang::csv::IO`
Error:    lang/php/stats/Unfriendly.rsc:1166:044: Undefined constructor `\map`
Error:    lang/php/stats/Unfriendly.rsc:0182:008: Undefined  `readCSV`
Error:    lang/php/stats/Unfriendly.rsc:1748:027: Undefined constructor `\set`
Error:    lang/php/stats/Unfriendly.rsc:1616:059: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:0048:040: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:0990:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0049:008: Undefined  `readCSV`
Error:    lang/php/stats/Unfriendly.rsc:1521:062: No definition for `symbol` in type `type[rel[str, str, str, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int]]`
Error:    lang/php/stats/Unfriendly.rsc:0795:008: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1496:030: Undefined constructor `\tuple`
Error:    lang/php/stats/Unfriendly.rsc:0582:001: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1447:026: Undefined constructor `\tuple`
Error:    lang/php/stats/Unfriendly.rsc:1166:076: Undefined constructor `Type::label`
Error:    lang/php/stats/Unfriendly.rsc:2118:008: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0037:007: Reference to name `lang::csv::IO` cannot be resolved
Error:    lang/php/stats/Unfriendly.rsc:1613:032: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:1166:067: No definition for `symbol` in type `type[map[str, tuple[int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int]]]`
Error:    lang/php/stats/Unfriendly.rsc:1122:011: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0773:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1935:008: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1447:021: Undefined constructor `\set`
Error:    lang/php/stats/Unfriendly.rsc:2126:008: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:2315:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0100:008: Undefined  `readCSV`
Error:    lang/php/stats/Unfriendly.rsc:1772:008: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0020:007: Reference to name `ValueIO` cannot be resolved
Warning:  lang/php/stats/Unfriendly.rsc:1648:048: Unused pattern variable `p`
Error:    lang/php/stats/Unfriendly.rsc:1608:008: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0584:001: Undefined  `readBinaryValueFile`
Warning:  lang/php/stats/Unfriendly.rsc:0434:022: Unused pattern variable `p`
Warning:  lang/php/stats/Unfriendly.rsc:1649:050: Unused pattern variable `p`
Error:    lang/php/stats/Unfriendly.rsc:0557:001: Undefined  `writeBinaryValueFile`
[INFO]    lang/php/stats/Unfriendly.rsc:2216:044: Pattern variable `v` has been declared outside pattern and its value will be used, add explicit declaration here if you want a new variable
[INFO]    lang/php/stats/Unfriendly.rsc:2219:047: Pattern variable `v` has been declared outside pattern and its value will be used, add explicit declaration here if you want a new variable
Error:    lang/php/stats/Unfriendly.rsc:0558:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:0791:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1521:036: Undefined constructor `\tuple`
Error:    lang/php/stats/Unfriendly.rsc:1496:056: No definition for `symbol` in type `type[rel[str, str, str, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int]]`
Warning:  lang/php/stats/Unfriendly.rsc:0427:024: Unused pattern variable `p`
Error:    lang/php/stats/Unfriendly.rsc:1117:013: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:0767:032: Undefined  `baseLoc`
Warning:  lang/php/stats/Unfriendly.rsc:0449:022: Unused pattern variable `p`
Error:    lang/php/stats/Unfriendly.rsc:1118:004: Undefined  `writeBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1128:016: Undefined constructor `\map`
[INFO]    lang/php/stats/Unfriendly.rsc:1716:021: Pattern variable `product` has been declared outside pattern and its value will be used, add explicit declaration here if you want a new variable
Error:    lang/php/stats/Unfriendly.rsc:1128:039: No definition for `symbol` in type `type[map[str, tuple[int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int]]]`
Error:    lang/php/stats/Unfriendly.rsc:0583:001: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1[826](https://github.com/cwi-swat/php-analysis/actions/runs/6271280617/job/17030583653#step:5:827):008: Undefined  `readBinaryValueFile`
Error:    lang/php/stats/Unfriendly.rsc:1128:048: Undefined constructor `Type::label`
Error:    lang/php/stats/Unfriendly.rsc:0173:040: Undefined  `baseLoc`
Error:    lang/php/stats/Unfriendly.rsc:1447:052: No definition for `symbol` in type `type[rel[str, str, str, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int]]`
Error:    lang/php/stats/Unfriendly.rsc:1603:013: Undefined  `baseLoc`
[INFO]    lang/php/stats/Unfriendly.rsc:2217:049: Pattern variable `v` has been declared outside pattern and its value will be used, add explicit declaration here if you want a new variable
Error:    lang/php/stats/Unfriendly.rsc:0772:059: Undefined  `baseLoc`
Warning:  lang/php/stats/Unfriendly.rsc:1650:046: Unused pattern variable `p`
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/analysis/cfg/BasicBlocks.rsc|
Warning:  lang/php/analysis/cfg/BasicBlocks.rsc:0018:000: Unused import of `lang::php::analysis::cfg::Label`
[INFO] Warnings and errors for |file:///home/runner/work/php-analysis/php-analysis/src/main/rascal/lang/php/util/Utils.rsc|
Error:    lang/php/util/Utils.rsc:0217:015: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0351:017: Undefined  `parserLoc`
Warning:  lang/php/util/Utils.rsc:0022:000: Unused import of `ValueIO`
Error:    lang/php/util/Utils.rsc:0365:056: Undefined  `parserLoc`
Error:    lang/php/util/Utils.rsc:0201:002: Undefined  `writeBinaryValueFile`
Error:    lang/php/util/Utils.rsc:0073:009: Undefined  `readTextValueString`
Error:    lang/php/util/Utils.rsc:0095:012: Undefined  `parserLoc`
Error:    lang/php/util/Utils.rsc:0457:002: Undefined  `writeBinaryValueFile`
Error:    lang/php/util/Utils.rsc:0377:038: Undefined  `phploc`
Error:    lang/php/util/Utils.rsc:0378:017: Undefined  `phploc`
Error:    lang/php/util/Utils.rsc:0407:004: Undefined  `writeBinaryValueFile`
Error:    lang/php/util/Utils.rsc:0499:002: Undefined  `writeBinaryValueFile`
Error:    lang/php/util/Utils.rsc:0300:014: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0353:046: Undefined  `parserLoc`
Error:    lang/php/util/Utils.rsc:0042:017: Undefined  `usePhpParserJar`
Error:    lang/php/util/Utils.rsc:0366:017: Undefined  `parserLoc`
Error:    lang/php/util/Utils.rsc:0192:014: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0336:014: Undefined  `logLevel`
Error:    lang/php/util/Utils.rsc:0219:002: Undefined  `writeBinaryValueFile`
Error:    lang/php/util/Utils.rsc:0065:139: Undefined  `parserWorkingDir`
Error:    lang/php/util/Utils.rsc:0362:013: Undefined  `parserLoc`
Error:    lang/php/util/Utils.rsc:0224:003: Undefined  `writeBinaryValueFile`
Error:    lang/php/util/Utils.rsc:0362:025: Undefined  `astToRascal`
Error:    lang/php/util/Utils.rsc:0022:007: Reference to name `ValueIO` cannot be resolved
Error:    lang/php/util/Utils.rsc:0302:008: Undefined  `readBinaryValueFile`
Error:    lang/php/util/Utils.rsc:0065:096: Undefined  `astToRascal`
Error:    lang/php/util/Utils.rsc:0270:014: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0399:014: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0354:017: Undefined  `parserLoc`
Error:    lang/php/util/Utils.rsc:0226:003: Undefined  `readBinaryValueFile`
Error:    lang/php/util/Utils.rsc:0216:014: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0230:002: Undefined  `readBinaryValueFile`
Error:    lang/php/util/Utils.rsc:0363:027: Undefined  `astToRascal`
Error:    lang/php/util/Utils.rsc:0065:041: Undefined  `parserMemLimit`
Error:    lang/php/util/Utils.rsc:0498:015: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0305:060: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0350:013: Undefined  `parserLoc`
Error:    lang/php/util/Utils.rsc:0199:015: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0375:017: Undefined  `phploc`
Error:    lang/php/util/Utils.rsc:0374:013: Undefined  `phploc`
Error:    lang/php/util/Utils.rsc:0455:015: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0365:068: Undefined  `astToRascal`
Error:    lang/php/util/Utils.rsc:0443:014: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0366:027: Undefined  `astToRascal`
Error:    lang/php/util/Utils.rsc:0058:017: Undefined variable, constructor or function `lang::php::util::Config::parserLoc`
Error:    lang/php/util/Utils.rsc:0210:014: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0022:007: Undefined module `ValueIO`
Error:    lang/php/util/Utils.rsc:0377:019: Undefined  `phploc`
Error:    lang/php/util/Utils.rsc:0403:015: Undefined  `readBinaryValueFile`
Error:    lang/php/util/Utils.rsc:0353:019: Undefined  `parserLoc`
Error:    lang/php/util/Utils.rsc:0019:007: Reference to name `lang::php::util::Config` cannot be resolved
Error:    lang/php/util/Utils.rsc:0137:005: Undefined  `includePhpDocs`
Error:    lang/php/util/Utils.rsc:0365:019: Undefined  `parserLoc`
Warning:  lang/php/util/Utils.rsc:0019:000: Unused import of `lang::php::util::Config`
Error:    lang/php/util/Utils.rsc:0445:001: Undefined  `writeBinaryValueFile`
Error:    lang/php/util/Utils.rsc:0198:014: Undefined  `parsedDir`
Error:    lang/php/util/Utils.rsc:0128:005: Undefined  `includeLocationInfo`
Error:    lang/php/util/Utils.rsc:0363:017: Undefined  `parserLoc`
Error:    lang/php/util/Utils.rsc:0042:043: Undefined  `phploc`
Error:    lang/php/util/Utils.rsc:0365:031: Undefined  `astToRascal`
Error:    lang/php/util/Utils.rsc:0019:007: Undefined module `lang::php::util::Config`
```

**Comment by jurgenvinju** [comment]
The current checker has `errorsAsWarnings` set to `true`, which is why the build does not fail on these reports. Later when we've all fixed them, let's set it to false to avoid those things creeping back in slowly.

**Comment by mahills** [comment]
Right. I'll take a look, I new about some of the warnings. For the errors, this is tricky, since these are references to values defined in `lang::php::util::Config` (which may be moved to `lang::php::config::Config` in the future, this is what we've done in Go AiR). However, these are install-specific, so we don't distribute this file. We instead distribute a template and make people add this file into their own project, either here or in a project that includes PHP AiR as a dependency. Is there a way to provide a version of this file to the build process, with the correct paths set? We don't have defaults for these.

**Comment by jurgenvinju** [comment]
proposal. we could use a JSON, XML, YAML, or MANIFEST file to configure php-analysis, in a similar way we use `PathConfig` instances to configure the Rascal compiler. We could even use PathConfig itself, and extend it with a number of fields?

* The `lang.manifest.IO` library binds the fields in a manifest file to the fields of an AST constructor. It also offers comma separated lists.
* `lang.json.IO`, and the json, yaml and xml variants all bind external data easily by mapping to data constructors as well.

And then we'd pass this configuration information down to all the analysis functions. We could have a default template file and users could use their own. Using the defaults of constructor keyword parameters, also sensible defaults can be programmed into the system.

That way we always have a Config module and the code would be ready for compilation.

**Comment by mahills** [comment]
I would be in favor of moving the config to a yaml file. The problem I ran into before was the following -- I need a way to load the file, but there doesn't seem to be a way for the project to figure out where it is. So, to load the file, I need a path, but to get the path, I would need to put it into a config file. If there is a way around this, that would be great -- I brought it up a while back (when I was starting to update this project) but couldn't really think of a good way around it.

**Comment by DavyLandman** [comment]
I think this has been solved.
